### ISSUE TITLE
White screen on first login on an account with two-factor auth

### ISSUE THREAD

**Issue by qrohlf** [rsms/fb-mac-messenger#299]
When attempting to launch/login for the first time using an account that has two-factor auth enabled, I get a blank white screen instead of the app:

![screen shot 2016-06-01 at 11 54 44 am](https://cloud.githubusercontent.com/assets/347189/15721836/a6021174-27ef-11e6-9dbd-efdf17f9cbc8.png)

Latest OSX.

**Comment by AdrianTP** [comment]
I also have this same problem. If I go to the menu and hit "Reload app from server", it briefly shows the following screen:

![image](https://cloud.githubusercontent.com/assets/1585276/15728144/5102162a-2820-11e6-9d0f-07ac0601818a.png)

...and then goes back to a white screen.

Also it takes an entire CPU core and gets super laggy to the point where it skips visibly while dragging the window around, but no other windows on my system skip like that.

**Comment by rmsa5** [comment]
Same issue here.

**Comment by skywinder** [comment]
In my case white scren, when I put wrong password. 2 fact auth not enabled

**Comment by hey-stefan** [comment]
Same issue, can't use the app.

**Comment by rsms** [comment]
This is pretty bad. I of course use 2FA, but can't reproduce the issue, so if anyone can repro this and submit a pull request, that would be really great.

**Comment by rsms** [comment]
What are you guys seeing in syslog/asl when this happens? (filter Messenger in Console.app)

**Comment by cdatwood** [comment]
6/17/16 10:38:27.139 AM Messenger[69476]: **\* WARNING: Method convertPointToBase: in class NSView is deprecated on 10.7 and later. It should not be used in new applications. 
6/17/16 10:39:17.712 AM Messenger[69476]: [evaluateJavaScript] Exception raised during -evaluateWebScript: main.js:272: TypeError: null is not an object (evaluating 'this.currentConversationItem().previousElementSibling')
6/17/16 10:39:17.712 AM Messenger[69476]: [evaluateJavaScript] Exception raised during -evaluateWebScript: main.js:283: TypeError: null is not an object (evaluating 'this.currentConversationItem().nextElementSibling')
6/17/16 10:41:28.093 AM Messenger[69641]: **\* WARNING: Method convertPointToBase: in class NSView is deprecated on 10.7 and later. It should not be used in new applications.

**Comment by rsms** [comment]
@cdatwood Thanks. None of that should cause the view to go blank. The JS exceptions should be just in the browser runloop call-out stack and should not affect other JS code in the same webview.

Could you please try the [new release 0.1.8](https://fbmacmessenger.rsms.me/) and see if the issue remains?

**Comment by rsms** [comment]
Here's a debug build of master that emits two additional log messages.

[Messenger-debug-2016-06-19-logwsredir.zip](https://github.com/rsms/fb-mac-messenger/files/322480/Messenger-debug-2016-06-19-logwsredir.zip)
1. `Delegating new-window navigation to workspace for URL "..." ("...")` — when a navigation happens inside the view of the kind that should open in a new window, and the app chooses to delegate the "open" action to macOS (which should open the default web browser for http/https URLs)
2. `Delegating forward navigation to workspace for URL "..." ("...")` — like the above, but logged for navigation actions that would have _replaced_ the contents of the web view, but are instead opened by the system.

You should see some no 1 messages, but should generally not see any no 2 messages.

If you could please try running this build and report back on what log messages you see in Console.app, that would be very helpful. Thank you!

**Comment by rsms** [comment]
See also #278

**Comment by amja** [comment]
Managed to solve this problem by disabling 2FA, logging in, and then reenabling it.

**Comment by DimitriBerman** [comment]
same issue here.

**Comment by VagishVela** [comment]
I tried with the latest version and I had the same issue.

**Comment by benomatis** [comment]
I don't have 2FA, yet the issue is there, I never got to make this app work for me, it's a shame as it looked so promising...

**Comment by joshvickerson** [comment]
This has come up before. I stumbled through it myself, as noted in #27. I believe this also generates a Facebook notification to "approve" the login, so if you go to Facebook in a browser you may have the option to approve the session. Then when you try to sign in again it'll let you in (I believe that's what happened to me before, but it's been so long I don't recall exactly).

**Comment by benomatis** [comment]
@joshvickerson yes, I've seen that being told by some, but I checked fb and didn't see anything that would resemble to an approval of some sort of this session.
